app/models/user_seen_subject.rb

```diff
@@ -5,14 +5,16 @@ class UserSeenSubject < ActiveRecord::Base
   validates_presence_of :user, :workflow
 
   def self.add_seen_subjects_for_user(user: nil, workflow: nil, subject_ids: nil)
-    uss = find_or_create_by!(user: user, workflow: workflow)
-    uss.add_subjects(subject_ids)
+    uss = where(user: user, workflow: workflow)
+    if uss.exists?
+      uss.add_subjects(subject_ids)
+    else
+      uss.create!(subject_ids: subject_ids)
+    end
   end
 
-  def add_subjects(subjects)
-    subject_ids_will_change!
-    subject_ids.concat(subjects)
-    save!
+  def self.add_subjects(subjects)
+    update_all(["subject_ids = uniq(subject_ids + array[?])", subjects])
```

Comment:
I'm slightly concerned that this method could be destructive if used on it's own without a scoped query. I.e. `UserSeenSubject.update_all(["subject_ids = uniq(subject_ids + array[?])", subjects])`.

Is it used anywhere else? If not can it be moved to where it's used in `#add_seen_subjects_for_user`.